.github/workflows/python-deps.yml

```diff
@@ -27,11 +27,22 @@ jobs:
     steps:
       - uses: actions/checkout@v2
 
+      - name: Install prerequisites
+        # brew can be finicky but it does not always means that the rest of the job will fail
+        run: |
+          # Needed for xmlsec
+          brew update -q && brew install libxml2 libxmlsec1 pkg-config || exit 0
+
       - name: Setup Python
         uses: ./.github/actions/setup-python
+        with:
+          # Non-default value since install-py-dev installs all requirements files
+          # We also want to bust the cache if the action and workflow change
+          cache-files-hash: ${{ hashFiles('requirements-*.txt', '.github/actions/setup-python/action.yml', '.github/workflows/python-deps.yml') }}
```

Comment:
The workflow and action rarely change but they should bust the cache to make sure we don't regress this (as it happened in PR #28352).

Every time one of the requirements file changes, we will bust the cache because we don't use `restore-key` in https://github.com/getsentry/sentry/blob/dc152314a48b84f4a1db1cb41efb6c345588ab92/.github/actions/setup-python/action.yml#L72-L77

@billyvg I think we should add a `restore-key`; what do you think?

Reply:
I think that's fine that cache busts when requirements changes. With `restore-key` there's no easy way to clean up the cache, so it can grow indefinitely.

Reply:
Sounds good.